File: examples/development/c/hello.c
/*
 * Build-only C example for MrHakOS.
 *
 * This is freestanding-style source: no stdio, no malloc, no host system calls.
 * It compiles to an object file today. It is not runnable inside MrHakOS until
 * the OS has a program loader and syscall/kernel-call ABI.
 */

int mrhakos_c_add(int a, int b) {
    return a + b;
}

int mrhakos_c_hello_status(void) {
    int total = 0;
    for (int i = 1; i <= 3; ++i) {
        total = mrhakos_c_add(total, i);
    }
    return total == 6 ? 0 : 1;
}

File: examples/development/python/hello.py
"""Python roadmap source for MrHakOS.

This file is intentionally source-only today. The repository can syntax-check it
with host Python, but MrHakOS cannot execute Python scripts until it has a
runtime substrate such as file I/O, memory allocation, syscalls, and either a
small interpreter or a port such as MicroPython.
"""


def main() -> int:
    total = sum((1, 2, 3))
    return 0 if total == 6 else 1


if __name__ == "__main__":
    raise SystemExit(main())

